adapters/play/src/main/scala/caliban/PlayWSMessage.scala

```diff
@@ -0,0 +1,7 @@
+package caliban
+
+import play.api.libs.json.JsObject
+
+final case class PlayWSMessage(id: String, messageType: String, payload: Option[JsObject]) {
+  lazy val request: Option[GraphQLRequest] = payload.flatMap(_.validate[GraphQLRequest].asOpt)
```

Comment:
Hint: `_.validate.asOpt` can just be `_.asOpt[GraphQLRequest]`

Reply:
Applied :)